src/mesh/dim2/adjacency_map/adjacency_map.cpp

```diff
@@ -0,0 +1,472 @@
+#include "enumerations/dimension.hpp"
+#include "enumerations/specfem_enums.hpp"
+#include "mesh/dim2/mesh.hpp"
+
+#include "index_mapping_from_adjacencies.cpp"
+#include <list>
+#include <stdexcept>
+#include <utility>
+
+// ====================================================================
+// conversions between edge and boundary enums.
+
+static inline specfem::enums::edge::type
+bdtype_to_edge(const specfem::enums::boundaries::type &bd) {
+  switch (bd) {
+  case specfem::enums::boundaries::TOP:
+    return specfem::enums::edge::type::TOP;
+  case specfem::enums::boundaries::LEFT:
+    return specfem::enums::edge::type::LEFT;
+  case specfem::enums::boundaries::RIGHT:
+    return specfem::enums::edge::type::RIGHT;
+  case specfem::enums::boundaries::BOTTOM:
+    return specfem::enums::edge::type::BOTTOM;
+  default:
+    return specfem::enums::edge::type::NONE;
+  }
+}
+static inline specfem::enums::boundaries::type
+edge_to_bdtype(const specfem::enums::edge::type &bd) {
+  switch (bd) {
+  case specfem::enums::edge::TOP:
+    return specfem::enums::boundaries::type::TOP;
+  case specfem::enums::edge::BOTTOM:
+    return specfem::enums::boundaries::type::BOTTOM;
+  case specfem::enums::edge::LEFT:
+    return specfem::enums::boundaries::type::LEFT;
+  case specfem::enums::edge::RIGHT:
+    return specfem::enums::boundaries::type::RIGHT;
+    break;
+  default:
+    // this should never be called
+    return specfem::enums::boundaries::type::RIGHT;
+  }
+}
+// ====================================================================
+// helper: given an edge, returns either the clockwise or counterclockwise
+// corner.
+static inline specfem::enums::boundaries::type
+edge_and_polarity_to_corner(const specfem::enums::edge::type &bd,
+                            const bool counterclockwise) {
+  switch (bd) {
+  case specfem::enums::edge::TOP:
+    return counterclockwise ? specfem::enums::boundaries::type::TOP_LEFT
+                            : specfem::enums::boundaries::type::TOP_RIGHT;
+  case specfem::enums::edge::BOTTOM:
+    return counterclockwise ? specfem::enums::boundaries::type::BOTTOM_RIGHT
+                            : specfem::enums::boundaries::type::BOTTOM_LEFT;
+  case specfem::enums::edge::LEFT:
+    return counterclockwise ? specfem::enums::boundaries::type::BOTTOM_LEFT
+                            : specfem::enums::boundaries::type::TOP_LEFT;
+  case specfem::enums::edge::RIGHT:
+    return counterclockwise ? specfem::enums::boundaries::type::TOP_RIGHT
+                            : specfem::enums::boundaries::type::BOTTOM_RIGHT;
+    break;
+  default:
+    // this should never be called
+    return specfem::enums::boundaries::type::RIGHT;
+  }
+}
+
+/**
+ * @brief calls set_as_boundary() for all boundary edges tagged by the
+ * mesh::mesh.
+ */
+static inline void boundarymark(
```

Comment:
you should follow the snake_case naming convention unless there's any specific reasons.